lib/pdn/Kconfig

```diff
@@ -31,6 +31,7 @@ config PDN_ESM_TIMEOUT
 
 config PDN_SYS_INIT
 	bool "Initialize during SYS_INIT"
+	depends on NRF_MODEM_LIB_SYS_INIT
```

Comment:
How about `default y if NRF_MODEM_LIB_SYS_INIT`
This is for easy of use. NRF_MODEM_LIB_SYS_INIT default 'y', then set PDN_SYS_INIT as 'y' all automatically.
If application explicitly set NRF_MODEM_LIB_SYS_INIT 'n', then  PDN_SYS_INIT is also 'n' and app needs to call pdn_init() after initialize modem_lib.

Reply:
Another comment, pdn_sys_init() could be saved and below logic should be part of pdn_init(), thinking about the case that application call pdn_init() explicitly with PDN_SYS_INIT 'n'

#if defined(CONFIG_PDN_DEFAULTS_OVERRIDE)
...
#endif

Then we can simply do sys_init as:

#if defined(CONFIG_PDN_SYS_INIT)
SYS_INIT(pdn_init, APPLICATION, CONFIG_PDN_INIT_PRIORITY);
#endif /* CONFIG_PDN_SYS_INIT */

Reply:
Do you mean moving:
```
#if defined(CONFIG_PDN_DEFAULTS_OVERRIDE)
	err = pdn_ctx_configure(0, CONFIG_PDN_DEFAULT_APN,
				CONFIG_PDN_DEFAULT_FAM, NULL);
	if (err) {
		LOG_ERR("Failed to configure default CID, err %d", err);
		return -1;
	}

#if defined(CONFIG_PDN_DEFAULT_AUTH_PAP) || defined(CONFIG_PDN_DEFAULT_AUTH_CHAP)
	/* +CGAUTH=<cid>[,<auth_prot>[,<userid>[,<password>]]] */
	BUILD_ASSERT(sizeof(CONFIG_PDN_DEFAULT_USERNAME) > 1, "Username not defined");

	err = pdn_ctx_auth_set(0, CONFIG_PDN_DEFAULT_AUTH,
			       CONFIG_PDN_DEFAULT_USERNAME,
			       CONFIG_PDN_DEFAULT_PASSWORD);
	if (err) {
		LOG_ERR("Failed to set auth params for default CID, err %d",
			err);
		return -1;
	}
#endif /* CONFIG_PDN_DEFAULT_AUTH_PAP || CONFIG_PDN_DEFAULT_AUTH_CHAP */
#endif /* CONFIG_PDN_DEFAULTS_OVERRIDE */
```
from `pdn_sys_init()` to `pdn_init()` so that initializing in `main()` or during `SYS_INIT` would have the same behavior?

Reply:
Yes, because when PDN_SYS_INIT is not set, we expect application to call pdn_init() explicitly, in which the overridden APN/credentials must be applied as well.

Reply:
OK, I have added a `default y if NRF_MODEM_LIB_SYS_INIT` and aligned the behavior when initializing manually or during during `SYS_INIT`.